test/modules/Modules.Test/TestResultCoordinator/TestReportUtilTest.cs

```diff
@@ -261,6 +265,48 @@ public void ParseReportMetadataList_ThrowExceptionWhenParseInvalidData()
             Assert.True(exceptionThrown);
         }
 
+        [Fact]
+        public void DirectMethodReportMetadataJsonDeserializeWithReceiverSourceTest()
+        {
+            const string SerializedTestJsonWithReceiverSource = "{ \"TestReportType\": \"DirectMethodReport\", \"SenderSource\": \"directMethodSender1.send\"," +
+            " \"ReceiverSource\": \"directMethodReceiver1.receive\", \"TolerancePeriod\": \"00:00:00.005\" }";
+            DirectMethodReportMetadata directMethodReportMetadata =
+                JsonConvert.DeserializeObject<DirectMethodReportMetadata>(SerializedTestJsonWithReceiverSource);
+            Assert.Equal("directMethodSender1.send", directMethodReportMetadata.SenderSource);
+            Assert.Equal(new TimeSpan(0, 0, 0, 0, 5), directMethodReportMetadata.TolerancePeriod);
+            Assert.Equal(TestReportType.DirectMethodReport, directMethodReportMetadata.TestReportType);
+            Assert.True(directMethodReportMetadata.ReceiverSource.HasValue);
+            directMethodReportMetadata.ReceiverSource.ForEach(x => Assert.Equal("directMethodReceiver1.receive", x));
+        }
+
+        [Fact]
+        public void DirectMethodReportMetadataJsonDeserializeWithoutReceiverSourceTest()
+        {
+            const string SerializedTestJsonWithoutReceiverSource = "{ \"TestReportType\": \"DirectMethodReport\", \"SenderSource\": \"directMethodSender1.send\", " +
+                "\"TolerancePeriod\": \"00:00:00.005\" }";
+            DirectMethodReportMetadata directMethodReportMetadata =
+                JsonConvert.DeserializeObject<DirectMethodReportMetadata>(SerializedTestJsonWithoutReceiverSource);
+            Assert.Equal("directMethodSender1.send", directMethodReportMetadata.SenderSource);
+            Assert.Equal(new TimeSpan(0, 0, 0, 0, 5), directMethodReportMetadata.TolerancePeriod);
+            Assert.Equal(TestReportType.DirectMethodReport, directMethodReportMetadata.TestReportType);
+            Assert.True(!directMethodReportMetadata.ReceiverSource.HasValue);
+        }
+
+        [Fact]
+        public void DirectMethodTestResultJsonSerializeTest()
+        {
+            DirectMethodTestResult directMethodTestResult = new DirectMethodTestResult(
```

Comment:
you should test serialize and deserialize in 2 tests; and you should not directly test JsonConvert.SerializeObjcet and JsonConvert.DeserializeObject.  You need to test the code in our codebase.